primefaces/src/main/resources/META-INF/resources/primefaces/fileupload/2-fileupload.js

```diff
@@ -386,6 +386,21 @@ PrimeFaces.widget.FileUpload = PrimeFaces.widget.BaseWidget.extend({
 
         this.jq.fileupload(this.ucfg);
     },
+    
+    /**
+     * @override
+     * @inheritdoc
+     */
+    destroy: function() {
+        try {
+            $(document).off("mouseover.fileupload mouseout.fileupload mousedown.fileupload mouseup.fileupload focus.fileupload blur.fileupload click.fileupload ", this.rowCancelActionSelector);
```

Comment:
may we also need the id here?

Reply:
agreed.

Reply:
I added the ID and also switch the `off()` to just use the name space which jQuery allows you to do and it removes all events with that namespace.  

```js
// Remove all events for a specific namespace
$(document).off('.yourNamespace');
```